lib/src/idx/ft/analyzer/mod.rs

```diff
@@ -127,62 +139,131 @@ impl Analyzer {
 			tfid.push((id, o.len() as TermFrequency));
 			osid.push((id, OffsetRecords(o)));
 		}
 		Ok((dl, tfid, osid))
 	}
 
-	fn analyze_content(&self, content: Vec<Value>, tks: &mut Vec<Tokens>) -> Result<(), Error> {
+	#[cfg_attr(not(target_arch = "wasm32"), async_recursion)]
+	#[cfg_attr(target_arch = "wasm32", async_recursion(?Send))]
+	async fn analyze_content(
+		&self,
+		ctx: &Context<'_>,
+		opt: &Options,
+		txn: &Transaction,
+		content: Vec<Value>,
+		tks: &mut Vec<Tokens>,
+	) -> Result<(), Error> {
 		for v in content {
-			self.analyze_value(v, tks)?;
+			self.analyze_value(ctx, opt, txn, v, tks).await?;
 		}
 		Ok(())
 	}
 
-	fn analyze_value(&self, val: Value, tks: &mut Vec<Tokens>) -> Result<(), Error> {
+	#[cfg_attr(not(target_arch = "wasm32"), async_recursion)]
+	#[cfg_attr(target_arch = "wasm32", async_recursion(?Send))]
+	async fn analyze_value(
+		&self,
+		ctx: &Context<'_>,
+		opt: &Options,
+		txn: &Transaction,
+		val: Value,
+		tks: &mut Vec<Tokens>,
+	) -> Result<(), Error> {
 		match val {
-			Value::Strand(s) => tks.push(self.analyze(s.0)?),
-			Value::Number(n) => tks.push(self.analyze(n.to_string())?),
-			Value::Bool(b) => tks.push(self.analyze(b.to_string())?),
+			Value::Strand(s) => tks.push(self.generate_tokens(ctx, opt, txn, s.0).await?),
+			Value::Number(n) => tks.push(self.generate_tokens(ctx, opt, txn, n.to_string()).await?),
+			Value::Bool(b) => tks.push(self.generate_tokens(ctx, opt, txn, b.to_string()).await?),
 			Value::Array(a) => {
 				for v in a.0 {
-					self.analyze_value(v, tks)?;
+					self.analyze_value(ctx, opt, txn, v, tks).await?;
 				}
 			}
 			Value::Object(o) => {
 				for (_, v) in o.0 {
-					self.analyze_value(v, tks)?;
+					self.analyze_value(ctx, opt, txn, v, tks).await?;
 				}
 			}
 			_ => {}
 		};
 		Ok(())
 	}
 
-	fn analyze(&self, input: String) -> Result<Tokens, Error> {
-		if let Some(t) = &self.t {
+	async fn generate_tokens(
+		&self,
+		ctx: &Context<'_>,
+		opt: &Options,
+		txn: &Transaction,
+		mut input: String,
+	) -> Result<Tokens, Error> {
+		if let Some(function_name) = &self.function {
+			let fns = format!("fn::{function_name}(\"{input}\");");
+			match path_like(&fns) {
+				Ok((_, func_value)) => {
+					let val = func_value.compute(ctx, opt, txn, None).await?;
+					if let Value::Strand(val) = val {
+						input = val.0;
+					} else {
+						return Err(Error::InvalidFunction {
+							name: function_name.to_string(),
+							message: "The function should return a string.".to_string(),
+						});
+					}
+				}
+				Err(e) => {
+					return Err(Error::InvalidFunction {
+						name: function_name.to_string(),
+						message: e.to_string(),
+					})
+				}
+			}
+		}
+		if let Some(t) = &self.tokenizers {
 			if !input.is_empty() {
 				let t = Tokenizer::tokenize(t, input);
-				return Filter::apply_filters(t, &self.f);
+				return Filter::apply_filters(t, &self.filters);
 			}
 		}
 		Ok(Tokens::new(input))
 	}
+
+	pub(crate) async fn analyze(
```

Comment:
@emmanuel-keller At a glance, I found it difficult to differentiate `analyze` vs the new `generate_tokens`! This is named `analyze` even though the core of the "analyzer" is implemented in the underlying `generate_tokens`.

Now, I take `analyze` as the core of the `search::analyze` function, where `generate_tokens` is obviously not exposed as a SurrealQL native function and is used internally by many analyze-related functions.
That said, I believe it would be easier to read if that's written as a code comment here! Maybe...

```suggestion
	/// Mainly used for exposing the analyzer as the SureralQL native function `search::analyze`
	pub(crate) async fn analyze(
```

Or if you have a better idea overall(rename functions along with code comments in other flavors..?) that would also be great!